### PR TITLE
[uart] FIFO reset signals to be self-cleared

### PR DESCRIPTION
**Pull request by eunchan**
This PR addresses the issue discussed in #4728 

I revised the UART design to self-clear the fifo reset signals in FIFO_CTRL register. Changed the HW access type to RW and uart_core module writes 0 back to fifo_ctrl.rxrst , fifo_ctrl.txrst when the values become 1.

Didn't make the fields external to avoid glitches on the line.

### CODE REVIEW

**Review by weicaiyang — APPROVED** [review]
LGTM

File: hw/ip/uart/data/uart.hjson [hunk 1]
```diff
@@ -179,13 +179,13 @@
       fields: [
         { bits: "0",
           swaccess: "wo",
-          hwaccess: "hro",
+          hwaccess: "hrw",
```
Comment (mdhayter) [inline hw/ip/uart/data/uart.hjson:182]: I think you can now remove hwqe: "true" (at least I can't see fifo_ctrl using qe anywhere, so I think it was just there for these two bits)

**Comment by eunchan** [comment]
Thanks for suggestion. It is removed now :)

**Comment by weicaiyang** [comment]
@eunchan 

Without this update, even it's not self-cleared, writing 1 to it will always trigger a reset. Looks like it's not need to write 0 after write 1. I assume writing 1 will reset fifo in scb and I don't always write 0 after that in test sequence, so I feel I probably don't need to update TB for this PR. LMK if I didn't understand correctly.
Screen Shot 2021-01-14 at 1 26 50 PM

/edascratch/weicai-opentitan/ot/update_edn/uart-sim-vcs/0.uart_fifo_reset/out

**Comment by mdhayter** [comment]
@weicaiyang the difference is if software reads it in the old way it will read a 1 (which makes it look like the fifo is in reset unless you read the docs very closely to understand the reset only happens as a pulse at the time of the write) but with the new way a read will see 0. (And in future if we needed multiple cycles to do the reset the software could potentially watch for the bit to clear to see the reset is done.)

**Comment by weicaiyang** [comment]
> @weicaiyang the difference is if software reads it in the old way it will read a 1 (which makes it look like the fifo is in reset unless you read the docs very closely to understand the reset only happens as a pulse at the time of the write) but with the new way a read will see 0. (And in future if we needed multiple cycles to do the reset the software could potentially watch for the bit to clear to see the reset is done.)

rxrst and txrst are tie to 0. Read should always return 0 for them. perhaps @eunchan can confirm this.
```
      addr_hit[7]: begin
        reg_rdata_next[0] = '0;
        reg_rdata_next[1] = '0;
        reg_rdata_next[4:2] = fifo_ctrl_rxilvl_qs;
        reg_rdata_next[6:5] = fifo_ctrl_txilvl_qs;
      end

```

**Comment by mdhayter** [comment]
Yes, I suppose the wo access would do that. In which case I'm not sure which version is better! But in neither do we need the sw to write zero.

**Comment by eunchan** [comment]
Ah.. yeah right. As previous implementation qualifies q with qe (and both are latched), you are right. Hum.. Then I think change this is irrelevant. We can drop this PR. Thanks for reminding the behavior. I totally forgot that :)

**Comment by weicaiyang** [comment]
> Ah.. yeah right. As previous implementation qualifies q with qe (and both are latched), you are right. Hum.. Then I think change this is irrelevant. We can drop this PR. Thanks for reminding the behavior. I totally forgot that :)

Thanks for confirming. sgtm.

**Review by msfschaffner — APPROVED** [review]

**Comment by silvestrst** [comment]
@eunchan does this mean that I should revert the added `0` after writing `1` in #4728 ?

**Comment by eunchan** [comment]
The previous design already does not require the software to write 0 after writing 1. So yes you can revert the added 0.

**Comment by silvestrst** [comment]
Thanks, will do!

**Comment by eunchan** [comment]
Closing as this PR is not needed.

**Pull request closed without merging**